plugins/ocp/ocp-nvme.c

```diff
@@ -0,0 +1,672 @@
+static __u8 scao_guid[C0_GUID_LENGTH]    = { 0xC5, 0xAF, 0x10, 0x28, 0xEA, 0xBF, 0xF2, 0xA4,
+        0x9C, 0x4F, 0x6F, 0x7C, 0xC9, 0x14, 0xD5, 0xAF };
+
+/* C3 Latency Monitor Log Page */
+#define C3_LATENCY_MON_LOG_BUF_LEN          0x200
+#define C3_LATENCY_MON_OPCODE               0xC3
+#define C3_LATENCY_MON_VERSION              0x0001
+#define C3_GUID_LENGTH                      16
+static __u8 lat_mon_guid[C3_GUID_LENGTH] = { 0x92, 0x7a, 0xc0, 0x8c, 0xd0, 0x84, 0x6c, 0x9c,
+        0x70, 0x43, 0xe6, 0xd4, 0x58, 0x5e, 0xd4, 0x85 };
+
+#define READ         0
+#define WRITE        1
+#define TRIM         2
+#define RESERVED     3
+
+typedef enum {
+    SCAO_PMUW               =  0,	/* Physical media units written */
+    SCAO_PMUR               = 16,	/* Physical media units read */
+    SCAO_BUNBR              = 32,	/* Bad user nand blocks raw */
+    SCAO_BUNBN              = 38,	/* Bad user nand blocks normalized */
+    SCAO_BSNBR              = 40,	/* Bad system nand blocks raw */
+    SCAO_BSNBN              = 46,	/* Bad system nand blocks normalized */
+    SCAO_XRC                = 48,	/* XOR recovery count */
+    SCAO_UREC               = 56,	/* Uncorrectable read error count */
+    SCAO_SEEC               = 64,	/* Soft ecc error count */
+    SCAO_EECE               = 72,	/* End to end corrected errors */
+    SCAO_EEDC               = 76,	/* End to end detected errors */
+    SCAO_SDPU               = 80,	/* System data percent used */
+    SCAO_RFSC               = 81,	/* Refresh counts */
+    SCAO_MXUDEC             = 88,	/* Max User data erase counts */
+    SCAO_MNUDEC             = 92,	/* Min User data erase counts */
+    SCAO_NTTE               = 96,	/* Number of Thermal throttling events */
+    SCAO_CTS                = 97,	/* Current throttling status */
+    SCAO_EVF                = 98,   /* Errata Version Field */
+    SCAO_PVF                = 99,   /* Point Version Field */
+    SCAO_MIVF               = 101,  /* Minor Version Field */
+    SCAO_MAVF               = 103,  /* Major Version Field */
+    SCAO_PCEC               = 104,	/* PCIe correctable error count */
+    SCAO_ICS                = 112,	/* Incomplete shutdowns */
+    SCAO_PFB                = 120,	/* Percent free blocks */
+    SCAO_CPH                = 128,	/* Capacitor health */
+    SCAO_NEV                = 130,  /* NVMe Errata Version */
+    SCAO_UIO                = 136,	/* Unaligned I/O */
+    SCAO_SVN                = 144,	/* Security Version Number */
+    SCAO_NUSE               = 152,	/* NUSE - Namespace utilization */
+    SCAO_PSC                = 160,	/* PLP start count */
+    SCAO_EEST               = 176,	/* Endurance estimate */
+    SCAO_PLRC               = 192,  /* PCIe Link Retraining Count */
+    SCAO_LPV                = 494,	/* Log page version */
+    SCAO_LPG                = 496,	/* Log page GUID */
+} SMART_CLOUD_ATTRIBUTE_OFFSETS;
+
+struct __attribute__((__packed__)) ssd_latency_monitor_log {
```

Comment:
FYI, __attribute__((__packed__)) will align all access to sizeof(1) so you might get compiler warnings for certain operation when this type is used. E.g when the size of the struct is not a multiple of the largest member and you 
alloc an array. Usually you want to add an aliment attribute along with the packed attribute.

See also https://stackoverflow.com/questions/8568432/is-gccs-attribute-packed-pragma-pack-unsafe

Reply:
In case the struct doesn't have holes due to natural alignment of the members, you don't gain anything from packed. Instead, it drops alignment requirements ('inner and outer') and the compiler will complain for certain things (passing
a member as argument in 'call by reference' situation IIRC) . As I think you don't have any holes I don't see a reason to have the packed. That was just an observation. If want to keep the packed I don't mind. My comment was triggered because we just had a bit of a discussion on this topic and I now I see this topic popping up everywhere.

Reply:
In this case, I think we can just keep packed, just so we stay consistent with the rest of the plugins, since a lot of them are also using it for their structs.